### PR TITLE
STM32WBA 1.3.1 Cube FW Updates Integration 

### PR DESCRIPTION
**Pull request by asm5878**
Integration of HAL/LL/BLE updates from STM32WBA Cube FW 1.3.1

### CODE REVIEW

File: stm32cube/stm32wbaxx/CMakeLists.txt [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Copyright (c) 2023 STMicroelectronics
+# Copyright (c) 2020 STMicroelectronics
```
Comment (erwango) [inline stm32cube/stm32wbaxx/CMakeLists.txt:1]: Not required, remove this change

File: stm32cube/stm32wbaxx/README [hunk 2]
```diff
@@ -35,6 +35,7 @@ License Link:
    https://opensource.org/licenses/BSD-3-Clause
 
 Patch List:
+--> please check that the following list is still valid:
```
Comment (erwango) [inline stm32cube/stm32wbaxx/README:38]: Remove this line

File: lib/stm32wba/hci/README [hunk 3]
```diff
@@ -100,6 +100,7 @@ License Link:
    opensource.org/license/mit
 
 Patch List:
+--> please check that the following list is still valid:
```
Comment (erwango) [inline lib/stm32wba/hci/README]: Remove this line

File: stm32cube/common_ll/README.rst [hunk 4]
```diff
@@ -15,20 +15,20 @@ stm32f0xx       1.11.4
...
 stm32l4xx       1.18.0
-stm32l5xx       1.5.0
+stm32l5xx       1.5.1
 stm32mp1xx      1.6.0
-stm32u5xx       1.4.0
-stm32wbaxx      1.2.0
-stm32wbxx       1.18.0
+stm32u5xx       1.5.0
+stm32wbaxx      1.3.1
+stm32wbxx       1.19.0
```
Comment (erwango) [inline stm32cube/common_ll/README.rst:32]: @marwaiehm-st Can you verify this is correct ?

File: lib/stm32wba/hci/app_conf.h [hunk 5]
```diff
@@ -449,7 +449,7 @@ typedef enum
 
 #if (USE_RADIO_LOW_ISR == 1)
 #define RADIO_SW_LOW_INTR_NUM               HASH_IRQn      /* Selected interrupt vector for 2.4GHz RADIO low ISR */
-#define RADIO_SW_LOW_INTR_PRIO              (15)           /* 2.4GHz RADIO low ISR priority */
+#define RADIO_SW_LOW_INTR_PRIO              (14)           /* 2.4GHz RADIO low ISR priority */
```
Comment (erwango) [inline lib/stm32wba/hci/app_conf.h:452]: Only this change is explained in the README.
Ideally provide one commit per change.

File: zephyr/module.yml [hunk 6]
```diff
@@ -4,16 +4,16 @@ build:
...
     description: "Binary Link Layer library for the STM32WBA Bluetooth subsystem"
   - path: stm32wba/lib/stm32wba_ble_stack_llo.a
-    sha256: d7f6053331539bc1353d2d9abf682caac89690bd480e1a637b29b94744132943
+    sha256: 8f0fb725f1c3facf0b53a67af5c001f16eb2e056f3d319f64c7f94799065e358
     type: lib
-    version: '1.2.0'
+    version: '1.3.1'
     license-path: zephyr/blobs/stm32wba/lib/license.md
-    url: https://github.com/STMicroelectronics/STM32CubeWBA/raw/v1.2.0/Middlewares/ST/STM32_WPAN/ble/stack/lib/stm32wba_ble_stack_llo.a
+    url: https://github.com/STMicroelectronics/STM32CubeWBA/raw/v1.3.1/Middlewares/ST/STM32_WPAN/ble/stack/lib/stm32wba_ble_stack_llo.a
```
Comment (erwango) [inline zephyr/module.yml:18]: Let's make it a dedicated commit

**Comment by asm5878** [comment]
Branch updated with the suggested changes.
Could you please review, thanks!

File: lib/stm32wba/hci/pta.h
Comment (erwango) [inline lib/stm32wba/hci/pta.h:1]: Please also add this file in `lib/stm32wba/hci/README`

**Review by erwango — CHANGES_REQUESTED** [review]
One last minor change and this will be fine to me

File: lib/stm32wba/hci/pta.h
Comment (asm5878) [reply lib/stm32wba/hci/pta.h:1]: PR updated with latest suggested modification.
Could you please review? Thanks

**Review by erwango — APPROVED** [review]

File: lib/stm32wba/hci/linklayer_plat.h [hunk 7]
```diff
@@ -180,4 +194,4 @@ extern void LINKLAYER_PLAT_DisableOSContextSwitch(void);
  */
 extern void LINKLAYER_PLAT_SCHLDR_TIMING_UPDATE_NOT(Evnt_timing_t * p_evnt_timing);
 
-#endif /* LINKLAYER_PLAT_H */
+#endif /* LINKLAYER_PLAT_H */
```
Comment (marwaiehm-st) [inline lib/stm32wba/hci/linklayer_plat.h:197]: To avoid this you should add newline at end of the file

File: lib/stm32wba/hci/ll_sys.h [hunk 8]
```diff
@@ -97,4 +98,11 @@ ll_sys_status_t ll_sys_dp_slp_exit(void);
...
 
-#endif /* LL_SYS_H */
+/**
+  * @brief  Get the number of concurrent state machines for the Link Layer
+  * @param  None
+  * @retval Supported number of concurrent state machines
+  */
+uint8_t ll_sys_get_concurrent_state_machines_num(void);
+
+#endif /* LL_SYS_H */
```
Comment (marwaiehm-st) [inline lib/stm32wba/hci/ll_sys.h:108]: Same

File: lib/stm32wba/hci/ll_sys_cs.c [hunk 9]
```diff
@@ -67,4 +67,4 @@ void ll_sys_enable_os_context_switch(void)
 void ll_sys_disable_os_context_switch(void)
 {
   LINKLAYER_PLAT_DisableOSContextSwitch();
-}
+}
```
Comment (marwaiehm-st) [inline lib/stm32wba/hci/ll_sys_cs.c:70]: same

File: lib/stm32wba/hci/pta.h [hunk 10]
```diff
@@ -0,0 +1,255 @@
...
+ * 			 			PTA_ERROR_SUCCESS : Otherwise.
+ */
+pta_error pta_set_mac_coex_priority(
+                uint32_t priority,
+                uint32_t priority_mask);
+#endif /* SUPPORT_MAC */
+/** @}
+*/
+#endif /* SUPPORT_PTA */
+#endif /* PTA_H_ */
```
Comment (marwaiehm-st) [inline lib/stm32wba/hci/pta.h:255]: Same

File: stm32cube/stm32wbaxx/drivers/include/stm32wbaxx_hal.h [hunk 11]
```diff
@@ -533,6 +523,7 @@ HAL_StatusTypeDef HAL_SYSCFG_GetConfigAttributes(uint32_t Item, uint32_t *pAttri
   * @}
   */
 
+
```
Comment (marwaiehm-st) [inline stm32cube/stm32wbaxx/drivers/include/stm32wbaxx_hal.h:526]: Remove the space

File: lib/stm32wba/hci/ll_sys_startup.c [hunk 12]
```diff
@@ -95,4 +95,4 @@ static void ll_sys_dependencies_init(void)
 
   /* Link Layer user parameters application */
   ll_sys_config_params();
-}
+}
```
Comment (marwaiehm-st) [inline lib/stm32wba/hci/ll_sys_startup.c:98]: same

File: lib/stm32wba/hci/ll_sys_startup.h [hunk 13]
```diff
@@ -26,4 +26,4 @@ void ll_sys_ble_cntrl_init(hst_cbk hostCallback);
 void ll_sys_mac_cntrl_init(void);
 void ll_sys_thread_init(void);
 
-#endif /* LL_SYS_STARTUP_H */
+#endif /* LL_SYS_STARTUP_H */
```
Comment (marwaiehm-st) [inline lib/stm32wba/hci/ll_sys_startup.h:29]: same

**Review by gautierg-st — CHANGES_REQUESTED** [review]
Only one file to remove + @marwaiehm-st comments, otherwise LGTM

**Comment by asm5878** [comment]
Dear All,
Here my new iteration.
Thanks for your feedback

**Comment by marwaiehm-st** [comment]
LGTM

**Review by marwaiehm-st — APPROVED** [review]

**Review by gautierg-st — APPROVED** [review]

**Review by erwango — APPROVED** [review]

**Pull request merged**